README.md

```diff
@@ -84,7 +84,11 @@ Parameters are provided to configure the behavior of the bridge. These parameter
  * (ROS 2) __num_threads__: The number of threads to use for the ROS node executor. This controls the number of subscriptions that can be processed in parallel. 0 means one thread per CPU core. Defaults to `0`.
  * (ROS 2) __min_qos_depth__: Minimum depth used for the QoS profile of subscriptions. Defaults to `1`. This is to set a lower limit for a subscriber's QoS depth which is computed by summing up depths of all publishers. See also [#208](https://github.com/foxglove/ros-foxglove-bridge/issues/208).
  * (ROS 2) __max_qos_depth__: Maximum depth used for the QoS profile of subscriptions. Defaults to `25`.
- * (ROS 2) __qos_reliability__: The default QoS reliability setting for subscriptions the bridge creates. Can be 'reliable', 'best_effort', or 'automatic'. Defaults to `automatic`.
+ * (ROS 2) __qos_reliability__: The default QoS reliability setting for subscriptions the bridge creates. Defaults to `automatic`.
+   * `reliable`: ALWAYS subscribe with a "reliable" QoS profile.
+   * `best_effort`: ALWAYS subscribe with a "best effort" QoS profile.
+   * `best_effort_if_volatile`: subscribe as "best effort" if all the participants are "volatile". If any are "transient_local", subscribe as "reliable".
```

Comment:
Why is it wrong to subscribe with "best effort" if any of the participants are "transient_local" ?

Reply:
I'm not sure I can fully answer this question but the following info may be helpful: https://docs.ros.org/en/rolling/Concepts/Intermediate/About-Quality-of-Service-Settings.html#qos-compatibilities

Reply:
It's a common pattern to publish a topic with "transient_local" durability whenever that value changes. For example a `/robot/healthy` Bool topic may be published only when the value changes from true<->false.

If the bridge subscribes to a `transient_local` topic with `best_effort` over a flakey connection. The historic `transient_local` message may be lost. These sorts of `transient_local` topics are typically not re-published so the bridge will never get the value.

`best_effort_if_volatile` is a useful default in this case but as mentioned, it's probably better to add a regex matcher param instead of this `qos_reliability` param.

i.e.

`reliable_qos_topic_list` - List of regular expressions ([ECMAScript grammar](https://en.cppreference.com/w/cpp/regex/ecmascript)) for topics to subscribe to as "reliable" (if publishers are all reliable). Defaults to [".*"].
`best_effort_qos_topic_list` - List of regular expressions ([ECMAScript grammar](https://en.cppreference.com/w/cpp/regex/ecmascript)) for topics to subscribe to as "best_effort". Defaults to ["(?!)"].

I might have a look at adding this today.

Reply:
👍 we'll take a look

> It's a common pattern to publish a topic with "transient_local" durability whenever that value changes. For example a /robot/healthy Bool topic may be published only when the value changes from true<->false.

FWIW my general advice on this is to to pick a cadence where you publish things like this even if they have not changed. It helps with data recording and debugging afterwards because you have the state information present without having to go back minutes or hours in history to lookup the topic value (which is hard in some workflows).